src/dmd/objc.d

```diff
@@ -499,22 +531,9 @@ extern(C++) private final class Supported : Objc
         id.objc.isExtern = true;
     }
 
-    override void deprecate(InterfaceDeclaration id) const
-    in
+    override const(char)* toPrettyChars(ClassDeclaration cd, bool qualifyTypes) const
     {
-        assert(id.classKind == ClassKind.objc);
-    }
-    do
-    {
-        // don't report deprecations for the metaclass to avoid duplicated
-        // messages.
-        if (id.objc.isMeta)
-            return;
-
-        id.deprecation("Objective-C interfaces have been deprecated");
-        deprecationSupplemental(id.loc, "Representing an Objective-C class " ~
-            "as a D interface has been deprecated. Please use "~
-            "`extern (Objective-C) extern class` instead");
+        return cd.parent.toPrettyChars(qualifyTypes);
```

Comment:
I'm not sure if this is the best way. Before this change, if a static method in an interface is not implemented in the class, the error message will include the name of the metaclass (something like `__anonclass_NSObject_Meta2`). This is an implementation detail which I don't want to expose. With this change, it will make sure to print the name of the class and not the metaclass.